paddle/fluid/jit/function.h

```diff
@@ -27,13 +27,16 @@ using Tensor = paddle::experimental::Tensor;
 
 class Function {
  public:
+  Function() = default;
```

Comment:
```suggestion
  Function() : engine_(nullptr) {};
```

不然， `engine_` 不会被初始化的。

Reply:
Would be better if you can also add tests to cover the newly added default constructor.

Reply:
done